tests/containers/helm.pm

```diff
@@ -42,24 +64,26 @@ sub run {
     assert_script_run("helm list");
 
     # Wait for deployment
-    script_retry("kubectl rollout status deployment/$self->{deployment_name}", delay => 30, retry => 5);
+    script_retry("kubectl rollout status deployment/$self->{deployment_name}", delay => 30, retry => 10);
+    assert_script_run("whereis kubectl");
     assert_script_run("kubectl describe deployment/$self->{deployment_name}");
     assert_script_run("kubectl get pods | grep $self->{deployment_name}");
     assert_script_run("kubectl describe services/$self->{deployment_name}");
 
     # Test
-    enter_cmd("kubectl port-forward  services/$self->{deployment_name} 10002:80 &");
-    script_retry("curl http://localhost:10002", delay => 30, retry => 5);
+    my $port = int(rand(100)) + 10000;
+    enter_cmd("kubectl port-forward  services/$self->{deployment_name} $port:80 &");
+    script_retry("curl http://localhost:$port", delay => 30, retry => 5);
 
     # Destroy the chart and k3s
     assert_script_run("helm delete $self->{deployment_name}");
-    uninstall_k3s();
+    uninstall_k3s() if $is_k3s;
 }
 
 sub post_fail_hook {
     my ($self) = @_;
     script_run("helm delete $self->{deployment_name}");
-    uninstall_k3s();
+    uninstall_k3s() if $self->{is_k3s};
```

Comment:
Why we need to remove `k3s` only? Sorry, if the question has been already answered. It just caught my attention.

Reply:
we uninstall k3s when k3s is installed previously. This step is not necessary but clean up a little the VM